### PR TITLE
[Feature](generic-aggregation) some prowork of generic aggregation

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

some prowork of generic aggregation

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.48 seconds
 stream load tsv:          421 seconds loaded 74807831229 Bytes, about 169 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230506101704_clickbench_pr_139353.html

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/olap/field.h [hunk 1]
```diff
@@ -642,22 +661,21 @@ class FieldFactory {
...
-        case OLAP_FIELD_AGGREGATION_BITMAP_UNION:
+        case FieldAggregationMethod::OLAP_FIELD_AGGREGATION_BITMAP_UNION:
             return new BitmapAggField(column);
-        case OLAP_FIELD_AGGREGATION_QUANTILE_UNION:
+        case FieldAggregationMethod::OLAP_FIELD_AGGREGATION_QUANTILE_UNION:
             return new QuantileStateAggField(column);
-        case OLAP_FIELD_AGGREGATION_UNKNOWN:
-            LOG(WARNING) << "WOW! value column agg type is unknown";
-            return nullptr;
+        case FieldAggregationMethod::OLAP_FIELD_AGGREGATION_UNKNOWN:
```
[the remark this answers is not in the document]
Comment (BiteTheDDDDt) [reply be/src/olap/field.h]: done

File: be/src/vec/olap/block_reader.cpp [hunk 2]
```diff
@@ -461,12 +461,22 @@ void BlockReader::_update_agg_value(MutableColumns& columns, int begin, int end,
...
+            } else {
+                function->add_batch_range(begin, end, place,
+                                          const_cast<const IColumn**>(&column_ptr), &_arena,
+                                          _stored_has_null_tag[idx]);
+            }
         }
 
         if (is_close) {
-            function->insert_result_into(place, *columns[_return_columns_loc[idx]]);
+            if (function->is_generic()) {
```
Comment (HappenLee) [inline be/src/vec/olap/block_reader.cpp:475]: the judge should be in `insert_result_into`

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**